File: PHP/Program/Clothing.php
<?php

/*
Saya Muhamad Tio Ariyanto [2201718] mengerjakan soal Latihan Praktikum 2
dalam mata kuliah DPBO untuk keberkahanNya saya tidak melakukan kecurangan
seperti yang telah dispesifikasikan. Aamiin
*/

require('Product.php');

class Clothing extends Product
{
    private $size = '';
    private $material = '';
    private $gender = '';

    // Constructor
    public function __construct($idProduct, $name, $brand, $price, $size, $material, $gender)
    {
        parent::__construct($idProduct, $name, $brand, $price);
        $this->size = $size;
        $this->material = $material;
        $this->gender = $gender;
    }

    // Setter dan getter
    public function setSize($size)
    {
        $this->size = $size;
    }

    public function getSize()
    {
        return $this->size;
    }

    public function setMaterial($material)
    {
        $this->material = $material;
    }

    public function getMaterial()
    {
        return $this->material;
    }

    public function setGender($gender)
    {
        $this->gender = $gender;
    }

    public function getGender()
    {
        return $this->gender;
    }
}
?>

File: CPP/Program/Clothing.cpp
/*
Saya Muhamad Tio Ariyanto [2201718] mengerjakan soal Latihan Praktikum 2
dalam mata kuliah DPBO untuk keberkahanNya saya tidak melakukan kecurangan
seperti yang telah dispesifikasikan. Aamiin
*/

// deklarasi library
#include <iostream>
#include <string>
#include "Product.cpp"

using namespace std;

class Clothing : public Product
{

    // atribut
private:
    string size;
    string material;
    string gender;

    // methods
public:
    Clothing()
    {
        this->size = "";
        this->material = "";
        this->gender = "";
    }

    Clothing(string size, string material, string gender)
    {
        this->size = size;
        this->material = material;
        this->gender = gender;
    }

    // getter
    string get_size()
    {
        return size;
    }
    string get_material()
    {
        return material;
    }
    string get_gender()
    {
        return gender;
    }

    // setter
    void set_size(string size)
    {
        this->size = size;
    }
    void set_material(string material)
    {
        this->material = material;
    }
    void set_gender(string gender)
    {
        this->gender = gender;
    }

    // destructor
    ~Clothing()
    {
    }
};